tools/slo-generator/tests/unit/test_compute.py

```diff
@@ -12,203 +12,101 @@
-        time_series_element = self.load_fixture(
-            filename=f'{cwd}/fixtures/time_series_proto.json', load_json=True)
-        time_series = [time_series_element]
-        expected_response = {
-            "next_page_token": next_page_token,
-            "time_series": time_series,
-        }
-        expected_response = metric_service_pb2.ListTimeSeriesResponse(
-            **expected_response)
-        channel = ChannelStub(responses=[expected_response] * nresp)
-        return channel
-
-    def setUp(self):
-        self.slo_config = self.load_fixture(
-            filename=f'{cwd}/fixtures/slo_linear.json',
-            load_json=True,
-            **FIXTURES_CONFIG)
-        self.slo_config_exp = self.load_fixture(
-            filename=f'{cwd}/fixtures/slo_exponential.json',
-            load_json=True,
-            **FIXTURES_CONFIG)
-        self.error_budget_policy = self.load_fixture(
-            filename=f'{cwd}/fixtures/error_budget_policy.json', load_json=True)
-        self.data = self.load_fixture(
-            filename=f'{cwd}/fixtures/slo_report.json', load_json=True)
-        self.timestamp = time.time()
-        self.good_event_count = 99
-        self.bad_event_count = 1
-        self.exporters = self.slo_config['exporters']
-
-    def test_compute_linear(self):
-        channel = self.make_grpc_stub(nresp=2 * len(self.error_budget_policy))
-        patch = mock.patch("google.api_core.grpc_helpers.create_channel")
-        with patch as create_channel:
-            create_channel.return_value = channel
-            compute(self.slo_config, self.error_budget_policy)
-
-    def test_compute_exponential(self):
-        channel = self.make_grpc_stub(nresp=2 * len(self.error_budget_policy))
-        patch = mock.patch("google.api_core.grpc_helpers.create_channel")
-        with patch as create_channel:
-            create_channel.return_value = channel
-            compute(self.slo_config_exp, self.error_budget_policy)
-
-    def test_compute_dummy_method(self):
-        results = compute(slo_config=self.slo_config,
-                          error_budget_policy=self.error_budget_policy,
-                          backend_method=dummy_slo_function)
-        results = list(results)
-        pprint.pprint(results)
-
-    def test_compute_dummy_obj(self):
-        results = compute(slo_config=self.slo_config,
-                          error_budget_policy=self.error_budget_policy,
-                          backend_obj=DummySLOBackend(),
-                          backend_method='dummy_slo_function')
-        results = list(results)
-        pprint.pprint(results)
-
-    @mock.patch(
-        "google.cloud.pubsub_v1.gapic.publisher_client.PublisherClient.publish")
-    @mock.patch("google.cloud.pubsub_v1.publisher.futures.Future.result")
+    @patch('google.api_core.grpc_helpers.create_channel',
+           return_value=mock_grpc_sd(2 * STEPS * len(SLO_CONFIGS_SD)))
+    def test_compute_stackdriver(self, mock):
```

Comment:
Rewrote test functions to allow testing all configs in the `samples/` folder. Ensures higher test coverage in general and  stronger tests.